### ISSUE TITLE
Cannot get gorm.ErrRecordNotFound when using pointer to slice

### ISSUE THREAD

**Issue by easonlin404** [jinzhu/gorm#2014]
I expect that can got `gorm.ErrRecordNotFound` If no record is found when using pointer to slice. And after trace code I noted that https://github.com/jinzhu/gorm/blob/master/callback_query.go#L87-L89 add   `isSlice` bool value to block error setting.

So how can I got no database record found when I using  pointer to slice? Thx.
```go
} else if scope.db.RowsAffected == 0 && !isSlice {
				scope.Err(ErrRecordNotFound)
}

```

### What version of Go are you using (`go version`)?
Latest

### Which database and its version are you using?
sqlite

### Please provide a complete runnable program to reproduce your issue. **IMPORTANT**

```go
package main

import (
	"github.com/jinzhu/gorm"
	_ "github.com/jinzhu/gorm/dialects/sqlite"
	"fmt"
)

type Product struct {
	gorm.Model
	Code string
	Price uint
}

func main() {
	db, _ := gorm.Open("sqlite3", "test.db")
	defer db.Close()

	// Migrate the schema
	db.AutoMigrate(&Product{})

	// Create
	db.Create(&Product{Code: "L1212", Price: 1000})

	var products []Product
	// No record can be found
	if err:= db.Where(map[string]interface{}{"code": ""}).Find(&products).Error;err!=nil {
		if err == gorm.ErrRecordNotFound {
			//expect getting ErrRecordNotFound error
			fmt.Print("ErrRecordNotFound")
		}
	}

}
```

**Comment by gm42** [comment]
This is totally unexpected; find on a slice should also return the error.

**Comment by easonlin404** [comment]
Merged #2015 ,close it.
